src/EnergyPlus/DaylightingDevices.cc

```diff
@@ -1620,6 +1614,164 @@ namespace DaylightingDevices {
+            ShowContinueError(
+                state, "Since the light shelf is above the window to which it is associated, the view factor of the window to the sky was reduced");
+            ShowContinueError(
+                state,
+                "and possibly also the view factor to the ground. Check you input and/or consider turning off autosizing of the view factors.");
+            leftoverViewFactor = 1.0 - viewFactorToShelf - viewFactorToGround;
+            if (leftoverViewFactor >= 0.0) {
+                viewFactorToSky = leftoverViewFactor;
+            } else {
+                viewFactorToSky = 0.0;
+                viewFactorToGround = 1.0 - viewFactorToShelf;
+                if (viewFactorToGround < 0.0) {
+                    viewFactorToGround = 0.0;
+                    viewFactorToShelf = 1.0;
+                }
+            }
+        } else { // At least part of the shelf is somewhere in the middle of the window so we need to split out the view factors
+            ShowContinueError(
+                state,
+                "Since the light shelf is neither fully above or fully below the window to which it is associated, the view factor of the window");
+            ShowContinueError(
+                state,
+                "to the ground and sky were both potentially reduced Check you input and/or consider turning off autosizing of the view factors.");
+            Real64 zShelfAvg;
+            if (((zShelfMin >= zWinMin) && (zShelfMax <= zWinMax)) || // Shelf does not go above or below the window
+                ((zShelfMin < zWinMin) && (zShelfMax > zWinMax))) {   // Shelf goes both above AND below the window
+                zShelfAvg = 0.5 * (zShelfMin + zShelfMax);
+            } else if (zShelfMin < zWinMin) { // Shelf goes partially below the window only
+                Real64 fracAbove = 0.0;
+                if (zShelfMax > zShelfMin) {
+                    fracAbove = (zShelfMax - zWinMin) / (zShelfMax - zShelfMin);
+                    if (fracAbove > 1.0) fracAbove = 1.0;
+                }
+                zShelfAvg = zWinMin + fracAbove * (zShelfMax - zWinMin);
+            } else { // (zShelfMax > zWinMax): Shelf goes partially above window
+                Real64 fracBelow = 0.0;
+                if (zShelfMax > zShelfMin) {
+                    fracBelow = (zWinMax - zShelfMin) / (zShelfMax - zShelfMin);
+                }
+                zShelfAvg = zWinMax - fracBelow * (zWinMax - zShelfMin);
+            }
+
+            // Find height ratio based on shelf average height
+            Real64 heightRatio;
+            if (zWinMax > zWinMin) { // Window has a positive height
+                heightRatio = (zShelfAvg - zWinMin) / (zWinMax - zWinMin);
+                heightRatio = min(heightRatio, 1.0);
+                heightRatio = max(heightRatio, 0.0);
+            } else { // Window does not have a positive height (not realistic) so set height ratio based on shelf location
```

Comment:
@jcyuan2020 Yes, this would be a weird scenario...I was trying to be as complete as possible, but it does have a built-in assumption that this is being used for a vertical wall.  I'm not sure that I have ever seen a lightshelf used with a skylight before.

Reply:
No problem---just trying to imagine a scenario where this block could be executed.